updateAnnouncements.py

```diff
@@ -0,0 +1,52 @@
+# Initial function:
+# Inputs: Github API credentials (not using for now)
+# call the github API to get all the comments for a certain issue on the hoth github
+# format the data into a json containing the data that we want
+# inject the json into a json file
+
+import requests
+import json
+import sys
+import os
+
+# User whitelist and issue number we are looking for
+user_whitelist = ['jodymlin', 'khxia', 'jakobreinwald', 'christinatong01', 'milesswu']
+# HOTH repository issues accessed via Github API
+repo_issues = "https://api.github.com/repos/uclaacm/hoth.uclaacm.com/issues"
+
+issue_num = '181'
+
+url = repo_issues + '/' + issue_num + '/' + 'comments'
+
+# Get given issue
+try:
+    announcements = requests.get(url)
+except requests.exceptions.RequestException as e:
+    raise SystemExit(e)
+print("Request completed with status code: ", announcements.status_code)
+
+# Convert announcements from Github API to json for parsing
+try:
+    announcements_json = announcements.json()
+except:
+    print("Error obtaining json from Github API return. ")
+valid_comments = []
+
+# Keep track of every comment from valid usernames
+try:
+    for index, element in enumerate(announcements_json):
+        if element['user']['login'] in user_whitelist:
+            body = element['body']
+            subject, partition, comment = body.partition('(Subject) ')
+            comment = {'id': index, 'subject': subject, 'body': comment, 'timestamp': element['created_at']}
+            valid_comments.append(comment)
+except:
+    print("Error parsing issue comments. ")
+
+# Insert all valid comment json objects into a file in reverse order
+filename = 'src/data/announcements.json'
+json_string = json.dumps(valid_comments[::-1], indent=4)
+with open(filename, 'w') as f:
```

Comment:
Let's also add a try-catch here, since writing to a file could fail.

Refer to this post: https://stackoverflow.com/questions/713794/catching-an-exception-while-using-a-python-with-statement